src/error.ts

```diff
@@ -744,21 +744,18 @@ const RETRYABLE_WRITE_ERROR_CODES = new Set<number>([
 ]);
 
 export function needsRetryableWriteLabel(error: Error, maxWireVersion: number): boolean {
-  if (maxWireVersion >= 9) {
-    // 4.4+ servers attach their own retryable write error
-    return false;
-  }
   // pre-4.4 server, then the driver adds an error label for every valid case
```

Comment:
Discussed in standup/tech chat we should bring this logic back and find out why our driver has a wire version value when handling a handshake error.